feat: Add configuration for reordering the prompt module and disabling default order (#171)
starship/starship: docs/config/README.md

### Options

| Variable       | Default | Description                                                        |
| -------------- | ------- | ------------------------------------------------------------------ |
| `add_newline`  | `true`  | Add a new line before the start of the prompt.                     |

> | `prompt_order` | `[]`    | Configure out custom prompt order overriding default prompt order. |

Reviewer: I think we may want to add the default `prompt_order` here, but it may be too big a list for the table.
Perhaps we could add a separate subb-heading to the "Prompt" heading specifically detailing the default prompt order and how changing it would work.

Reviewer 2: @matchai That was originally in the PR, but I voiced concerns about having that exposed separately (it means that we'd have to mirror changes in this doc as well every time we make a change, and we've been making a lot recently, with two or three PRs in the last 24 hours that would have needed to remember to make changes there).

Would it be possible to insert something here that generates the order directly from the code, so that we don't have to keep them in sync?

Reviewer 2: Perhaps a really simple solution would be to include a link to the GitHub source code (with the correct line number) for anyone who wants to know what the default order is.

Reviewer: > Perhaps a really simple solution would be to include a link to the GitHub source code (with the correct line number)

I'm in favor of this 👍

Author: Main issue for linking out line number is changing line number since above `DEFAULT_PROMPT_ORDER` we have `ALL_MODULE` const and import statement which needs to be updated out for each addition of module since each modules need to be added over `ALL_MODULE` for custom prompt order to work as well as we may add new imports. Maybe adding subheading as orginally in the PR and commenting `DEFAULT_PROMPT_ORDER` code with information that if someone change this value they need to also update config docs

Reviewer: I think that's reasonable. 👍

> | `prompt_order` | `[]`    | Configure out custom prompt order overriding [default prompt order](#default-prompt-order). |

Reviewer: I think this may be a slightly clearer way of explaining this option.

```suggestion
| `prompt_order` | [link](#default-prompt-order)    | Configure the order in which the prompt modules appear. |
```

### Options

| Variable                | Default | Description                                                          |
| ----------------------- | ------- | -------------------------------------------------------------------- |
| `add_newline`           | `true`  | Add a new line before the start of the prompt.                       |
| `prompt_order`          | `[]`    | Configure out custom prompt order modifying default order            |

> | `disable_default_order` | `false` | Disable default config order and only enable out custom prompt order |

Reviewer: I don't think we really need the `disable_default_order` configuration option.
The default value of `prompt_order` could be retrieved from [PROMPT_ORDER](https://github.com/starship/starship/blob/master/src/print.rs#L10-L25), and overriding it in configuration would disable the default prompt order.

Please let me know if you have other functionality in mind around that configuration. 🙂

Author: I have nothing special functionality in mind. But what about adding some custom module `languages` which will add all supported languages instead of adding all languages in config file. So, starship can easily support new languages module which are added in future by default on new version of starship without modifying custom config file. User can also add one by one languages by themselves.

Author: If `language` custom module is good custom module for addition then it may be added in another PR separately rather then implementing in this PR.

Reviewer 2: I think it would be a great idea, but getting the implementation right would be a challenge--you're essentially talking about implementing an extensions client for starship. We probably won't be ready to do that for some time.

# Disable the newline at the start of the prompt
add_newline = false
prompt_order=["rust","line_break","package","line_break","character"]
```

> ### default_prompt_order

Reviewer 2: I'm not 100% sure about exposing this in the docs right now: we currently have two other PRs in the pipeline to add new modules and several additional issues about modules beyond that. If we throw this in now, we'll have to make sure that this and `print.rs` stay in sync.

I do think it's a good idea, but we might want to save it for 1.0.

Author: I have removed expose of default_prompt_order since I also believe it is not good idea to add this currently with other PR open. Can implement this in near future after addition of all language module

Author: But after merging of this PR other PR open needs to add out added module in `ALL_MODULE` const in `print.rs` to support out custom prompt order for those module but need not to update config README

# Disable the newline at the start of the prompt
add_newline = false
# Overwrite a default_prompt_order and  use custom prompt_order
prompt_order=["rust","line_break","package","line_break","character"]
```

### Default prompt order

> ```default_prompt_order``` is used as ```prompt_order``` when config have no options named as ```prompt_order``` or when you have set ```prompt_order = []```

Reviewer: A little bit of nitpicking around how this option is documented.

```suggestion
The ```default_prompt_order``` configuration option is used to define the order in which modules are shown in the prompt. If no ```default_prompt_order``` is provided, the default is as shown:
```
